samples/socket-api/wasm-src/udp_client.c

```diff
@@ -73,6 +73,7 @@ main(int argc, char *argv[])
                    (struct sockaddr *)&server_address, &serverlen);
 
     if (ret > 0) {
+        buffer[ret] = '\0';
```

Comment:
Will ret be equal to sizeof(buffer)? If yes, buffer[ret] will cause illegal memory access.
How about change to:
```
char msg[64];
snprintf(msg, sizeof(msg), "[Client] Buffer recieved: %%.%ds\n", ret);   => construct string "%.ns", n == ret
printf(msg, buffer);
```

Reply:
according to code `core/iwasm/libraries/lib-socket/src/wasi/wasi_socket_ext.c`,  it * returns the number of bytes read or -1 for errors.*.